### PR TITLE
TASK-55628 Fix Copying enrolled node in publication lifecycle

### PR DESCRIPTION
**Pull request by boubaker**
Prior to this change, when copy/paste a content that is enrolled in Publication lifecycle, the property 'publication:liveRevision' (of type REFERENCE) references the original content published version. Consequently when attempting to delete original content from JCR, a ReferentialIntegrityException is thrown which will prevent content deletion. This change will enroll again the copied content inside the publication lifecycle to cleanup references to original node and will make the copied content as Draft, then change the state again to the original content's state (Published if it was 'Published', 'Staged', if it was 'Staged'...).

For previously copied contents and placed in Trash, we will have to add a specific workaround to allow deleting it (since the reference already exists and used the old 'Paste' algorithm that didn't cleaned Publication Information).
In fact, a previously published webcontent, which was copied at least once, will not be able to be deleted since the new copied nodes holds a reference to the same live Revision as the original. This workaround will attempt to delete those references by publishing again the copies (if no modification was already made by author user), else this will make the copied content as 'Draft' only without a referenced 'Live revision'. This will lead to unpublish some copied content as consequence that should be acceptable comparing to the fact that the original content can't be deleted anymore.

### CODE REVIEW

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java [hunk 1]
```diff
@@ -401,22 +420,24 @@ private void processRemoveNode(String nodePath, Node node, Event<?> event, boole
       node.remove();
       parentNode.save();
     } catch (VersionException ve) {
+      LOG.warn("Error deleting node", ve);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:423]: Should be LOG.error ? 
```suggestion
      LOG.error("Error deleting node", ve);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java [hunk 2]
```diff
@@ -401,22 +420,24 @@ private void processRemoveNode(String nodePath, Node node, Event<?> event, boole
...
       parentNode.save();
     } catch (VersionException ve) {
+      LOG.warn("Error deleting node", ve);
       uiApp.addMessage(new ApplicationMessage("UIPopupMenu.msg.remove-verion-exception", null,
                                               ApplicationMessage.WARNING));
 
       uiExplorer.updateAjax(event);
       return;
     } catch (ReferentialIntegrityException ref) {
+      LOG.warn("Error deleting node", ref);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:430]: Should it be LOG.error ?
```suggestion
      LOG.error("Error deleting node", ref);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java [hunk 3]
```diff
@@ -401,22 +420,24 @@ private void processRemoveNode(String nodePath, Node node, Event<?> event, boole
...
-      uiApp
-      .addMessage(new ApplicationMessage(
+      uiApp.addMessage(new ApplicationMessage(
                                          "UIPopupMenu.msg.remove-referentialIntegrityException", null,
                                          ApplicationMessage.WARNING));
 
       uiExplorer.updateAjax(event);
       return;
     } catch (ConstraintViolationException cons) {
+      LOG.warn("Error deleting node", cons);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:440]:
```suggestion
      LOG.error("Error deleting node", cons);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java [hunk 4]
```diff
@@ -425,17 +446,16 @@ private void processRemoveNode(String nodePath, Node node, Event<?> event, boole
       uiExplorer.updateAjax(event);
       return;
     } catch (LockException lockException) {
+      LOG.warn("Error deleting node", lockException);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:449]:
```suggestion
      LOG.error("Error deleting node", lockException);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java [hunk 5]
```diff
@@ -425,17 +446,16 @@ private void processRemoveNode(String nodePath, Node node, Event<?> event, boole
...
                                               ApplicationMessage.WARNING));
 
       uiExplorer.updateAjax(event);
       return;
     } catch (Exception e) {
-      if (LOG.isErrorEnabled()) {
-        LOG.error("an unexpected error occurs while removing the node", e);
-      }
-      JCRExceptionManager.process(uiApp, e);
+      LOG.warn("an unexpected error occurs while removing the node", e);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:456]:
```suggestion
      LOG.error("an unexpected error occurs while removing the node", e);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java [hunk 6]
```diff
@@ -589,28 +609,28 @@ private static void processPaste(ClipboardCommand currentClipboard, String destP
         thumbnailService.processRemoveThumbnail(srcThumbnailNode);
       }
     } catch (ConstraintViolationException ce) {
-      LOG.debug("Error pasting document from {} to {}", srcPath, destPath, ce);
+      LOG.warn("Error pasting document from {} to {}", srcPath, destPath, ce);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java:610]:
```suggestion
      LOG.error("Error pasting document from {} to {}", srcPath, destPath, ce);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java [hunk 7]
```diff
@@ -589,28 +609,28 @@ private static void processPaste(ClipboardCommand currentClipboard, String destP
...
-      LOG.debug("Error pasting document from {} to {}", srcPath, destPath, ce);
+      LOG.warn("Error pasting document from {} to {}", srcPath, destPath, ce);
       uiApp.addMessage(new ApplicationMessage("UIPopupMenu.msg.current-node-not-allow-paste", null,
           ApplicationMessage.WARNING));
 
       uiExplorer.updateAjax(event);
       return;
     } catch (VersionException ve) {
-      LOG.debug("Error pasting document from {} to {}", srcPath, destPath, ve);
+      LOG.warn("Error pasting document from {} to {}", srcPath, destPath, ve);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java:617]:
```suggestion
      LOG.error("Error pasting document from {} to {}", srcPath, destPath, ve);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java [hunk 8]
```diff
@@ -589,28 +609,28 @@ private static void processPaste(ClipboardCommand currentClipboard, String destP
...
       uiApp.addMessage(new ApplicationMessage("UIPopupMenu.msg.copied-node-in-versioning", null,
           ApplicationMessage.WARNING));
 
       uiExplorer.updateAjax(event);
       return;
     } catch (ItemExistsException iee) {
-      LOG.debug("Error pasting document from {} to {}", srcPath, destPath, iee);
+      LOG.warn("Error pasting document from {} to {}", srcPath, destPath, iee);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java:624]:
```suggestion
      LOG.error("Error pasting document from {} to {}", srcPath, destPath, iee);

```

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java [hunk 9]
```diff
@@ -589,28 +609,28 @@ private static void processPaste(ClipboardCommand currentClipboard, String destP
...
       uiApp.addMessage(new ApplicationMessage("UIPopupMenu.msg.paste-node-same-name", null,
           ApplicationMessage.WARNING));
 
       uiExplorer.updateAjax(event);
       return;
     } catch (LoginException e) {
-      LOG.debug("Error pasting document from {} to {}", srcPath, destPath, e);
+      LOG.warn("Error pasting document from {} to {}", srcPath, destPath, e);
```
Comment (rdenarie) [inline core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/PasteManageComponent.java:631]:
```suggestion
      LOG.error("Error pasting document from {} to {}", srcPath, destPath, e);

```

**Review by rdenarie — COMMENTED** [review]
I made suggestion for log levels. 
As modification you add are related to exception, should we use error log level instead of warn ?

**Review by rdenarie — COMMENTED** [review]
In addition, Sonar show code smells about unused imports. Can you remove it ?

**Comment by boubaker** [comment]
> In addition, Sonar show code smells about unused imports. Can you remove it ?

Done

**Review by rdenarie — APPROVED** [review]

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java (hunk 1 above)
Comment (boubaker) [reply core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:423]: As discussed, will preserve the WARN logging level here and I will attempt to clarify more the difference between both levels in Best Practices.

File: core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java (hunk 2 above)
Comment (boubaker) [reply core/webui-explorer/src/main/java/org/exoplatform/ecm/webui/component/explorer/rightclick/manager/DeleteManageComponent.java:430]: As discussed, will preserve the WARN logging level here and I will attempt to clarify more the difference between both levels in Best Practices.

**Pull request merged**